### ISSUE TITLE
FileChooser directory names differ from system language

### ISSUE THREAD

**Issue by Bios-Marcel** [JFormDesigner/FlatLaf#178]
Most things in the file chooser are translated, but the directory names of the users data directories are wrong.
So when your system is german for example, it should be `Dokumente`, instead it is `Documents` though.
The Windows L&F does this right. I haven't found any of these values in the UI manager though.
I couldn't really figure out where these names come from. However, Metal does the same mistake.
Since the FlatFileChooserUI extends the Metal one, that might be why it doesn't work out of the Box.

**EDIT**
I did find a bug report that describes a slightly different bug:
https://bugs.openjdk.java.net/browse/JDK-4423439
Might be related, don't know.

Either way, what's even more weird, is that on different machines with equal language settings, we sometimes do and sometimes don't have the problem. Also, there are multiple entries for `Documents` and only the one displayed under the system drive (C:/) is english. The shortcuts at the top are translated correctly. The same occurs with `C:/Users`, which should be `C:/Benutzer`. This is pretty weird.

**Comment by DevCharly** [comment]
Thx for reporting.

The localized file/directory names come from Windows OS via method `sun.awt.shell.Win32ShellFolder2.getDisplayName()`, which is (or should be) invoked from `javax.swing.filechooser.FileSystemView.getSystemDisplayName()`.

Seems that it is necessary to set UI property `FileChooser.useSystemExtensionHiding` to `true` to enable this feature. This is what the Windows LaF does. This flag also hides known file extensions, if enabled in Windows Explorer.

I'll add this flag for the next release. In the meantime you can enable it in your app with:

~~~java
UIManager.put( "FileChooser.useSystemExtensionHiding", true );
~~~

**Comment by Bios-Marcel** [comment]
Thanks for the info.

**Comment by Bios-Marcel** [comment]
So, this is quite funny and confusing. Using the Flag you mentioned, it works as expected. But shouldn't that flag also be hiding extension? I can still see extensions on all filetypes. Is that how it's supposed to be?

**Comment by DevCharly** [comment]
No, it should not hide all extensions. Only extensions that Windows "knows". Whatever that means... 
On my system it hides e.g. `.txt`, `.jpg`, `.m4v`, ...

Also following option must be enabled (in Windows Explorer Options dialog) for hiding known extensions:

![grafik](https://user-images.githubusercontent.com/5604048/94191513-e6036180-fead-11ea-9e7d-6bb7369e0548.png)

**Comment by Bios-Marcel** [comment]
Oh I see. I've disabled that, as it's quite useful to see all file extensions imho ^^

Thanks for letting me know :)

**Comment by DevCharly** [comment]
fixed in 0.43
